src/stores/A_store.js

```diff
@@ -153,6 +153,45 @@ class A_store extends ReduceStore {
                 }
             }
 
+            case ActionTypes.DSTARLITE: {
+                const {player, end, pathGrid} = state;
+                const {grid} = pathGrid;
+                const DSL = new Dstarlite(grid[player.y][player.x], grid[end.y][end.x], Cell.heuristic);
+                DSL.beginPathfinding();
+
+                return {...state, DSL, stage: 'DSTAR_STEP'};
+            }
+
+            case ActionTypes.DSTAR_STEP: {
+                const {DSL, seenCells, revealedGrid, end} = state;
+                let {pathGrid} = state;
+                const nextLocation = DSL.nextStep();
+                let stage;
+                let visibleCells;
+
+                if (nextLocation.x !== end.x || nextLocation.y !== end.y) {
```

Comment:
i feel like this logic would've been easier to follow if you did the opposite and checked the reset case first

Reply:
🐇